src/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs

```diff
@@ -121,14 +121,22 @@ public abstract class XmlSerializerImplementation
     /// </devdoc>
     public class XmlSerializer
     {
+#if uapaot
```

Comment:
That is generally the rule but there are exceptions to that rule similar to what you need this for so it can be in public in the implementation if it is needed for things like reflection.